pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -231,9 +231,15 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
     }
 
     if (request.has(Broker.Request.QUERY_OPTIONS)) {
-      Map<String, String> queryOptions = getOptionsFromRequest(request, Broker.Request.QUERY_OPTIONS);
-      LOGGER.debug("Query options are set to: {} for request {}: {}", queryOptions, requestId, query);
-      brokerRequest.setQueryOptions(queryOptions);
+      Map<String, String> queryOptionsFromRequest = getOptionsFromRequest(request, Broker.Request.QUERY_OPTIONS);
```

Comment:
its probably better to create another map queryOptions.

queryOptions = new HashMap();

if(queryOptionsFromRequest != null) {
 queryOptions.addAll(queryOptionsFromRequest)
}
if(queryOptionsFromSQL != null) {
 queryOptions.addAll(queryOptionsFromSQL)
}

brokerRequest.setQueryOptions(queryOptions);

Reply:
refactored